File: SDKs/nexmo.com/voice/1.3.9/python/src/sdk/models/shared/getcallsresponse.py
import dataclasses
from typing import Optional
from dataclasses_json import dataclass_json
from sdk import utils
from ..shared import getcallresponse as shared_getcallresponse


@dataclass_json
@dataclasses.dataclass
class GetCallsResponseEmbedded:
    r"""GetCallsResponseEmbedded
    A list of call objects. See the [get details of a specific call](#getCall) response fields for a description of the nested objects
    """
    
    calls: Optional[list[shared_getcallresponse.GetCallResponse]] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('calls') }})
    

@dataclass_json
@dataclasses.dataclass
class GetCallsResponseLinksSelf:
    href: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('href') }})
    

@dataclass_json
@dataclasses.dataclass
class GetCallsResponseLinks:
    self: Optional[GetCallsResponseLinksSelf] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('self') }})
    

@dataclass_json
@dataclasses.dataclass
class GetCallsResponse:
    embedded: Optional[GetCallsResponseEmbedded] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('_embedded') }})
    links: Optional[GetCallsResponseLinks] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('_links') }})
    count: Optional[int] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('count') }})
    page_size: Optional[int] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('page_size') }})
    record_index: Optional[int] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('record_index') }})
    

File: SDKs/nexmo.com/voice/1.3.9/java/lib/src/main/java/openapisdk/models/shared/GetCallsResponse.java
package openapisdk.models.shared;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonInclude.Include;

public class GetCallsResponse {
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("_embedded")
    public GetCallsResponseEmbedded embedded;
    public GetCallsResponse withEmbedded(GetCallsResponseEmbedded embedded) {
        this.embedded = embedded;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("_links")
    public GetCallsResponseLinks links;
    public GetCallsResponse withLinks(GetCallsResponseLinks links) {
        this.links = links;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("count")
    public Long count;
    public GetCallsResponse withCount(Long count) {
        this.count = count;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("page_size")
    public Long pageSize;
    public GetCallsResponse withPageSize(Long pageSize) {
        this.pageSize = pageSize;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("record_index")
    public Long recordIndex;
    public GetCallsResponse withRecordIndex(Long recordIndex) {
        this.recordIndex = recordIndex;
        return this;
    }
}